### PR TITLE
fix: amend breaking changes and reinstate constraints

### LINKED ISSUE #511: input type="search" isn't considered of role "searchbox"
Previously considered the following code as of role "searchbox" but it is now considered as "textbox"

```html
<input type="search" />
```

Now I have to add a `list` attribute to have the input with the implicit role of "searchbox".

```html
<input type="search" list="foo" />
```

Shouldn't this be considered a breaking change? I have an entire project that relies on the previous behavior.

As from MDN [Technical summary](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#technical_summary)

> **Implicit ARIA role**
> 
> - type=search
>     - with no list attribute: searchbox
>     - with list attribute:combobox

This behavior is happening since v5.2.0

### LINKED ISSUE #512: Image and combobox are not being detected by the getByRole.
### Describe the bug
Image tagged element and combobox are not being detected by the getByRole query.

Aria-query previous working version - 5.1.3
Aria-query upgraded version version - 5.2.1

``

Previously before the '**aria-query**' upgrade, getByRole query will detect the above element. But after the upgrade this query is not able to detect the above element.

We have many test cases that are breaking due to above issue . This behavior is happening since v5.2.1. Please fix the issue above issue.

### PR DESCRIPTION
**Pull request by jlp-craigmorten**
Fixes #511 
Fixes #512 
Relates to https://github.com/testing-library/dom-testing-library/issues/1239

**Description**

#447 brought in a number of breaking changes to this repo and was released under a minor version, namely:

1. Removal of the `constraints` array within related concepts `attributes`.
2. Additions / updates according to the unstable [W3C Working Draft for HTML-AAM](https://www.w3.org/TR/html-aam-1.0/).

This PR looks to partially revert some of these changes (with a few other unrelated corrections and additions) to provide an upgrade path to downstream packages, using the following as references:

- https://www.w3.org/TR/html-aria/#docconformance
- https://www.w3.org/TR/wai-aria-1.2/#role_definitions
- https://www.w3.org/TR/html-aam-1.0/#html-element-role-mappings
- https://github.com/A11yance/aria-query/tree/v5.1.3

In some places I have left the HTML-AAM additions where they could be considered non-breaking additive changes [in line with discussion here](https://github.com/A11yance/aria-query/issues/498#issuecomment-1579855088). Open to changes where feel should keep some of the HTML-AAM changes. I think the key is to resolve (1) which has broken the likes of dom-testing-library.

**Summary of changes**

- Added `@babel/node` as a dev dep so `npm run output_as_hack` can be used.
- Updated browserslist db as prompted by console, running `npx update-browserslist-db@latest`.
- Removed `section` element association from `section` role as unable to find any references in any stable specs nor the HTML-AAM Working Draft.
- Removed `aria-controls` constraint on `combobox` role to fix #511. This is new in HTML-AAM Working Draft, but not in WAI-ARIA 1.2 nor HTML-ARIA W3C Recommendations.
- Correct to `select` element constraints involving `multiple`. I believe the spec has been misinterpreted as `multiple` being included in the condition regarding a number value. [It is a boolean attribute](https://html.spec.whatwg.org/multipage/form-elements.html#attr-select-multiple) so have corrected constraint wording.
- Reinstated the `constraints` arrays for attributes, reverting it's removal in #447. In some places additional attribute `constraints` were added to align to the spec.
- Addition of `'ancestor table element has treegrid role'` constraint to `gridcell` role: "role=[gridcell](https://www.w3.org/TR/html-aria/#index-aria-gridcell) if the ancestor table element is exposed as a role=grid or treegrid". See https://www.w3.org/TR/html-aria/#docconformance.
- Reinstated the `img` element specific conditions on `alt`.
- Removed `aria-multiselectable` constraint on `datalist` for `listbox` role. This is new in HTML-AAM Working Draft, but not in WAI-ARIA 1.2 nor HTML-ARIA W3C Recommendations.
- Reinstated the ARIA `listitem` related concept to the `menuitem` role as per https://www.w3.org/TR/wai-aria-1.2/#menuitem.
- Removed constraints on `progressbar` role for `progress` element. This is new in HTML-AAM Working Draft, but not in WAI-ARIA 1.2 nor HTML-ARIA W3C Recommendations.
- Amended the `button` related concept for the `switch` role. The related concept is for the ARIA `button` not HTML, see https://www.w3.org/TR/wai-aria-1.2/#switch.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
     "prepublishOnly": "npm run build",
     "pretest": "npm run lint:fix && npm run flow",
     "test": "npm run jest",
-    "test:ci": "npm run jest -- --ci --runInBand",
+    "test:ci": "npm run jest -- --ci --runInBand --watch",
```
Comment (cmorten) [inline package.json:17]: 🤦 

```suggestion
    "test:ci": "npm run jest -- --ci --runInBand",
```

**Review by jessebeach — APPROVED** [review]
I super appreciate the time and care put into this. Let's get this in and put out a minor release to revert the breakage. This is helping me better understand how downstream packages are using this package.

**Comment by MatanBobi** [comment]
Hi @jessebeach!
Next time if you want us (testing-library) to run tests prior to release (v6), please don't hesitate to reach out :)

**Comment by jessebeach** [comment]
Thank you @MatanBobi! I will take you all up on this offer.

**Pull request merged**